docs/tutorial.rst

```diff
@@ -1,4 +1,209 @@
 Tutorial
 ========
 
-*Coming soon!*
+In this document, we are going to develop an application that will
+convert a given time (and date) between two time zones.
+The user will enter a date and a time,
+and select two time zones from a list of all available time zones,
+one for the source location and one for the destination location.
+
+.. note::
+
+   Time zones are represented in "region/location" format,
+   as in "Australia/Tasmania".
+   While most such codes have two components,
+   some contain only one (like "UTC"),
+   and some contain more than two (like "America/North_Dakota/New_Salem".
+   Also note that spaces in region and location names are replaced
+   with underscores.
+   See the `Wikipedia list`_ for more information.
+
+
+Prerequisites
+-------------
+
+First of all, we have to install the dependencies.
+These are: *clastic* (obviously) and *dateutil*.
+Note that the PyPI package name for *dateutil* is *python-dateutil*::
+
+  pip install clastic python-dateutil
+
+
+
+Getting started
+---------------
+
+Here is the initial version of our application that only displays
+the form but doesn't handle the submitted data:
+
+.. code-block:: python
+
+   import os
+
+   from clastic import Application
+   from clastic.render import AshesRenderFactory
+   from dateutil import zoneinfo
+
+
+   def home():
+       zone_info = zoneinfo.get_zonefile_instance()
+       zone_names = zone_info.zones.keys()
+       entries = ((zone.split("/")[-1], zone) for zone in zone_names)
```

Comment:
```suggestion
       entries = [(zone.split("/")[-1], zone) for zone in zone_names]
```

I tend to prefer list comps when memory isn't an issue. They're more introspectable when debugging and also faster in most cases (less generator state for the runtime to setup and maintain).

Reply:
I agree that a list comprehension is better here. I must have done it like that at first, that would also have been my preference. I guess I had used `sorted` on entries at this point, but later moved it to the next comprehension and while removing it here I must have unintentionally converted it into a generator comp. Anyway, I changed it into a list comp.